### DISCUSSION TITLE
Need help configuring Mailserver as Relay

### DISCUSSION THREAD

**Discussion by tedstriker** [docker-mailserver/docker-mailserver#3770] [Q&A]
Hello,

I'm trying to use the mail server as a relay for outgoing mail for my domain. But unfortunately I can't get it to work, because
I couldn't find the - what I think - necessary configuration attributes in form of environment variables, config files, etc. if they exist.

### Environment
The container with the "to be relay mail server" runs on a hosted server with a fixed IP. Lets call it external mail server for better distinction.
It will receive Mails _from_ *@myowndomain.com from an internal mail server, where all accounts and aliases are configured.

So the flow would be: my email -> internal mail server -> external mail server -> destination mail server

### Goals
Here's what I'm trying to achieve

- [x] I want the internal mail server to forward all outgoing mail to the external mail server
I know how to configure it, so that's already possible
- [ ] The DMS should reject unencrypted (non-TLS) connections. Maybe it can be enforced on port 25 as well.
- [ ] The internal server should authenticate against the external server with a username and password when using it as relay. This user may have a mailbox, but it would not be necessary as it is not supposed to receive any email.
- [ ] The external server should only deliver mails from _*@myowndomain.com_ to _*@anyotherdomainintheworld.com_.

### Bonus goals
- [] If I could use the external server also to receive mail for _*@myowndomain.com_ and forward them to the internal server.
- [] In case the internal server is unreachable, the external server stores mails until the internal server becomes reachable again.

### What I got so far...
- only SMTP is running using `SMTP_ONLY=1`
- created a user, but no authentication is requested
- LetsEncrypt certificate is configured and can be used (confirmed with swaks)
### The struggle
All the time I'm thinking "It can't be that complicated", probably it's just a few variables to be set, but for my life I can't figure out.
Is this even possible with Docker Mail Server?

Any help is appreciated
Thank you.

**Comment by polarathene** [comment]
> * The DMS should reject unencrypted (non-TLS) connections. Maybe it can be enforced on port 25 as well.

Port 25 is for delivery. Mail inbound is checked against an account to deliver to, if there's a valid recipient it's accepted, otherwise it's rejected.

If this server is never meant to receive mail, don't expose port 25.

Port 587 is for StartTLS mail submission. This is what your relay should be using to authenticate against a DMS user account on the relay DMS. You configure credentials and it will relay the mail for you after successful authentication.

I believe we have port 587 configured to reject if TLS cannot be established via StartTLS. However if you want to enforce it, you should be using port 465 instead which is implicit TLS similar to HTTPS.

---

> The internal server should authenticate against the external server with a username and password when using it as relay.
> This user may have a mailbox, but it would not be necessary as it is not supposed to receive any email.

This is a null mail client. There are similar names for it.

We partially support this with `SMTP_ONLY=1` which disables Dovecot (Mail storage), but presently there is no support for creating an account. This is similar to the relay submission mentioned above (your external MTA).

Technically if you have no use for the internal DMS for anything else, the applications that would mail it could connect to the external DMS over port 465 with credentials (_either with the same username/password, or individual credentials to better isolate a compromised account_).

A common setup is to have local services submit mail via DMS on port 465, and have DMS use a third-party like SendGrid as the relayhost (_DMS sends outbound mail to the relayhost with credentials configured_). In your description, you're using DMS as a relayhost (external MTA).

---

> The external server should only deliver mails from `*@myowndomain.com` to `*@anyotherdomainintheworld.com`.

We have `SPOOF_PROTECTION=1` to restrict an account from what mail they can send as. The default (`SPOOF_PROTECTION=0`) allows any DMS account to submit mail as anybody (_including domains you don't actually have the right to send as_).

You can enable `SPOOF_PROTECTION=1`, but you will want to relax the restriction. I think our documentation is lacking on how to do that, but it enables the `smtpd_sender_login_maps` Postfix setting in `main.cf`. There is a specific lookup table appended to that setting by the feature in addition to general alias lookup tables, but I think this was not implemented well (_it's in need of a refactor, there's just a lack of time to spend on it_).

We have a separate lookup table though that would be useful here, a regexp alias (`pcre:/etc/postfix/regexp`), [this one is covered in our docs](https://docker-mailserver.github.io/docker-mailserver/v13.2/config/user-management/#configuring-regexp-aliases) and all you need to do is add the `postfix-regexp.cf` to your Docker config volume (_it'll be copied to that internal location during container startup_). You'd have something like:

```
/(.*@myowndomain.com)$/ ${1}
```

---

> If I could use the external server also to receive mail for `*@myowndomain.com` and forward them to the internal server.

I believe we have support for Dovecot to be set to an external instance instead. However you'd need to keep any accounts in sync between the two that it's kinda like setting up two DMS instances anyway..?

One other option might be to just have the external/public server proxy to the internal one? (inbound traffic flows from external server to internal DMS, and outbound traffic for internal DMS flows through external server)

---

>  In case the internal server is unreachable, the external server stores mails until the internal server becomes reachable again.

Ok, so this is really about the DMS instance on the external server being your primary instance, with DNS records to direct mail to it from that domain.

The internal instance could instead sync / pull mail from that external instance, or perhaps via aliases forward it to an internal non-public domain?

I think in that instance, it may have Postfix try to send the mail from external DMS to internal DMS, and when not reachable, it'll be kept in a queue to try again with exponential backoff.

---

> created a user, but no authentication is requested

Oh alright, you already went with `SMTP_ONLY=1`. Like mentioned above, this is not quite full-featured, the contributor didn't document it well.

You can use LDAP or look through related issues on the feature for how other users were manually setting up a user account.

---

> Is this even possible with Docker Mail Server?

I don't think I see this setup discussed often.

But a null mail client instance yes. A relayhost instance yes. The other concerns perhaps are just approaching it the wrong way?

**Comment by dfoxg** [comment]
@tedstriker did your setup now works? Would you like to share your configs? I want to have the same setup (home mailserver behind a dynamic ip already receives all my mails - now i want to use my other, external mailserver with a static ip for sending my private mails).

**Reply by polarathene** [reply]
If you just want the external mailserver to relay mail that's simpler than what was asked here.

Look at the rewritten relay host docs (_select the edge version, it'll be part of v14 release_), that should have everything you need to have your home mailserver to authenticate on port 587 to submit mail for relay.

For better delivery you could also just relay through SendGrid or similar services instead, probably cheaper (_especially for personal only usage_) and they have established solid sender reputation already.

**Reply by dfoxg** [reply]
Thanks for the quick answer! Sadly, at least my current setup doesnt work.

I configured my local docker-mailserver (=home server) as following:
```
DEFAULT_RELAY_HOST=<my remote docker mailserver>
RELAY_USER=<my remote user>
RELAY_PASSWORD=<my remote user password>
```

When i send a mail from my local domain i get following logs on my home server:
```
Feb 11 13:08:56 mail postfix/submissions/smtpd[1030]: connect from pixel7.wg.devices.home[100.64.0.2]
Feb 11 13:08:56 mail postfix/submissions/smtpd[1030]: Anonymous TLS connection established from pixel7.wg.devices.home[100.64.0.2]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature ECDSA (P-256) server-digest SHA256
Feb 11 13:08:56 mail postfix/submissions/smtpd[1030]: 959221D00F6F: client=pixel7.wg.devices.home[100.64.0.2], sasl_method=PLAIN, sasl_username=<my local email>
Feb 11 13:08:56 mail postfix/sender-cleanup/cleanup[1035]: 959221D00F6F: message-id=<EAFA848F-6F06-4933-8279-8F9B4C4A06B9@<my local docker mailserver>>
Feb 11 13:08:56 mail postfix/sender-cleanup/cleanup[1035]: 959221D00F6F: replace: header MIME-Version: 1.0 from pixel7.wg.devices.home[100.64.0.2]; from=<<my local email>> to=<<test email adress from another mail provider>> proto=ESMTP helo=<[127.0.0.1]>: MIME-Version: 1.0
Feb 11 13:08:56 mail opendkim[738]: 959221D00F6F: DKIM-Signature field added (s=mail, d=<my local docker mailserver>)
Feb 11 13:08:56 mail postfix/qmgr[862]: 959221D00F6F: from=<<my local email>>, size=727, nrcpt=1 (queue active)
Feb 11 13:08:56 mail postfix/submissions/smtpd[1030]: disconnect from pixel7.wg.devices.home[100.64.0.2] ehlo=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=6
Feb 11 13:08:57 mail postfix/smtpd-amavis/smtpd[1042]: connect from localhost[127.0.0.1]
Feb 11 13:08:57 mail postfix/smtpd-amavis/smtpd[1042]: 735881D02755: client=localhost[127.0.0.1]
Feb 11 13:08:57 mail postfix/cleanup[1043]: 735881D02755: message-id=<EAFA848F-6F06-4933-8279-8F9B4C4A06B9@<my local docker mailserver>>
Feb 11 13:08:57 mail postfix/qmgr[862]: 735881D02755: from=<<my local email>>, size=1443, nrcpt=1 (queue active)
Feb 11 13:08:57 mail amavis[937]: (00937-01) Passed CLEAN {RelayedOpenRelay}, [100.64.0.2]:45742 <<my local email>> -> <<test email adress from another mail provider>>, Queue-ID: 959221D00F6F, Message-ID: <EAFA848F-6F06-4933-8279-8F9B4C4A06B9@<my local docker mailserver>>, mail_id: Xzyt-4PYBkjj, Hits: 1.776, size: 1220, queued_as: 735881D02755, 699 ms
Feb 11 13:08:57 mail postfix/smtp-amavis/smtp[1036]: 959221D00F6F: to=<<test email adress from another mail provider>>, relay=127.0.0.1[127.0.0.1]:10024, delay=0.91, delays=0.17/0.03/0.01/0.7, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 735881D02755)
Feb 11 13:08:57 mail postfix/qmgr[862]: 959221D00F6F: removed
Feb 11 13:09:04 mail postfix/smtp[1044]: Trusted TLS connection established to mail.<my remote docker mailserver>[178.238.226.67]:25: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256
Feb 11 13:09:04 mail postfix/smtp[1044]: 735881D02755: to=<<test email adress from another mail provider>>, relay=mail.<my remote docker mailserver>[178.238.226.67]:25, delay=6.6, delays=0.02/0.02/6.6/0.05, dsn=5.7.1, status=bounced (host mail.<my remote docker mailserver>[178.238.226.67] said: 554 5.7.1 <<test email adress from another mail provider>>: Relay access denied (in reply to RCPT TO command))
Feb 11 13:09:04 mail postfix/cleanup[1043]: 1D4411D0275E: message-id=<20240211120904.1D4411D0275E@mail.<my local docker mailserver>>
Feb 11 13:09:04 mail postfix/bounce[1054]: 735881D02755: sender non-delivery notification: 1D4411D0275E
Feb 11 13:09:04 mail postfix/qmgr[862]: 1D4411D0275E: from=<>, size=3480, nrcpt=1 (queue active)
Feb 11 13:09:04 mail postfix/qmgr[862]: 735881D02755: removed
```

my remote docker mailserver as following logline:
```
Feb 11 12:09:04 mailserver-docker-mailserver-bd488fcc8-7lgbq postfix/smtpd[5741]: NOQUEUE: reject: RCPT from <my ip>: 554 5.7.1 <<test email adress from another mail provider>>: Relay access denied; from=<<my local email>> to=<<test email adress from another mail provider>> proto=ESMTP helo=<mail.<my local docker mailserver>>
```

Do i have to change the settings on my remote docker mailserver so that it can act as a relay server for my local docker mailserver?

**Reply by polarathene** [reply]
I think your config settings did not match the docs advice at the time which was `DEFAULT_RELAY_HOST` needed `RELAY_HOST` to be specified. There were some bugs but they've been resolved for upcoming v14 release.

You should be fine with your setup as it is with v14. You could try with our current `:edge` build if you like to confirm?

I also updated the relay host docs page quite a bit, along with the related ENV on the ENV docs page. Give those a look if you like, just visit the docs and at the top left select the version as `edge`.

**Reply by dfoxg** [reply]
I updated my local mailserver to the latest :edge-Version, but still got the same errors. I think i have to configure my remote docker mailserver to allow relays for my own domain (like a allowlist), but have no idea where i can do that. The Relay-Docs are only focused for the sending part, not receiving/forwarding a relayed mail.

**Reply by polarathene** [reply]
> I updated my local mailserver to the latest :edge-Version, but still got the same errors

Odd, I implemented the fixes and verified the configuration as documented in `:edge` works between two DMS instances (at least locally).

> I think i have to configure my remote docker mailserver to allow relays for my own domain (like a allowlist), but have no idea where i can do that.

No, all that changes is the mail is sent out to a different MTA that will handle the direct connection for delivery over port 25.

You should have `DEFAULT_RELAY_HOST=mail.example.com:587` to connect to that instance of DMS as if you were using Thunderbird (or any other mail client) to submit mail directly. It's the same process of authenticating with credentials for a user that instance is aware of. DMS accounts aren't tied to the email sender address you use (_unless using `SPOOF_PROTECTION=1`_).

- Presently you cannot use port 465 for relay, DMS needs additional work to handle that for you (_alternatively you can bypass the feature and provide your own config directly in `postfix-main.cf` / `postfix-master.cf`_).
- Likewise port 25 is not for authentication, only delivery or forwarding (via alias). So you must configure for port 587 when DMS is the relay host.

---

> The Relay-Docs are only focused for the sending part, not receiving/forwarding a relayed mail.

Sorry I don't follow?

They document how to configure DMS to relay mail through another mail server (MTA). That could be another instance of DMS or a mail service like SendGrid.

- Any sending is related to you submitting mail to the DMS with that relay config, with `DEFAULT_RELAY_HOST` all outbound mail will be sent through the configured relay host to handle delivery for you on your behalf.
- I believe for the third MTA in this chain (_the one that is receiving the mail_), you'll need to have DNS authorizing the relay to send on your behalf, such as your SPF record. That way when it arrives at the third-party MTA, and they do the usual security checks to trust the mail is actually from you, those checks pass.

Perhaps the DNS part is an issue for you? I could cover that too, usually it's assumed a third-party service like SendGrid is being used where the user would be reading instructions from that service and stuff like this is covered by the service, the user just needs to know the config part for DMS specifically which is what we've covered.

---

> Logs provided: `Relay access denied`

This is a different kind of relay. The kind that arrives on port 25 and wants to delivery to an address DMS doesn't manage.

DMS does not support `relay_domains`, which handles relaying mail from third-parties (_which could be another mail server of yours too technically_) delivering to port 25.

- Without a relay host, if you receive mail only to send it out to someone else directly, this is known as relaying.
- When DMS instead sends mail to a MTA that handles the direct delivery, that MTA is a relay host (it does the relaying).

We only support the latter. You can do the former if you'd like to configure that yourself as per Postfix docs. If you're only interested in relaying your own mail sent, you'd use a relay host.

You can create an alias that forwards mail to an external address, that is supported as it's a fixed delivery address (_you can have a wildcard catch-all alias, but it's going to a single outbound address(s)_), not something the third-party can control (_which would make you an open relay if too permissive, and that is bad_).

**Reply by dfoxg** [reply]
thanks for that answer!

i´m had only exposed port 465 through my haproxy like this:
```
listen lb_smtp_465_587
        bind <ip>:465
        bind <ip>:587
        mode tcp
        option tcplog

        tcp-request content reject if { src -f /media/shared/configs/haproxy/blacklisted.ips }

        server l0004 10.0.0.4:30465 send-proxy
```

I updated the logic (exposed 587 to port 587 of the remote docker mailserver and now im getting a other kind of error message:

```
Feb 17 09:41:36 mailserver-docker-mailserver-bd488fcc8-7lgbq postfix/submission/smtpd[1709673]: NOQUEUE: reject: RCPT from <myip>: 554 5.7.1 <myip>: Client host rejected: Access denied; from=<my private home server email> to=<the test mail from another mail provider> proto=ESMTP helo=<mail.<home domain>>
```

Do you have any idea?

**Reply by dfoxg** [reply]
I also just noticed that i´m using "SPOOF_PROTECTION=1" on my remove mailserver - would it help to create a mail account on the remote server, that matches local mail? For example I have the email local@mylocal.de on my local mailserver, i could create the exact same mail on the remote server (and use that credentials for the relay login).

**Reply by polarathene** [reply]
> i´m had only exposed port 465 through my haproxy like this:

Relay to port 465 on remote is not supported by DMS yet.

> Do you have any idea?

Probably `SPOOF_PROTECTION=1` as you mentioned, disabling that fixes it?

> would it help to create a mail account on the remote server, that matches local mail?

The feature only lets you send mail from the login email address. So if you have `local@mylocal.de` as the sender address, `SPOOF_PROTECTION=1` expects that will be used as the login credentials too. If you don't have other users than yourself, you could just disable this feature and then any login account can use any sender address, doesn't matter what it is or who owns it.

When you create an account with `setup email add` or `setup alias add`, DMS will think it manages that account and accept mail for it. If that remote DMS ever needs to send mail out to that address it may think it already looks after it and thus not send but attempt delivery instead.

---

BTW, on the edge docs, the examples tutorial section has updated the "mailserver with proxy" page. It focuses on Traefik but if you are not sure about your proxy config with PROXY protocol, it should be better covered there now.

**Reply by dfoxg** [reply]
I disabled `SPOOF_PROTECTION` but still got the same error. Is there are possibility to increase the log level for postfix? The environment variable `LOG_LEVEL=debug` has no effect on postfix.

Currently following `smtpd_recipient_restrictions` is set  (i think this is the default value, i didnt change anything there) on the remote server:
```
root@mailserver-docker-mailserver-78c7976486-q6dvq:/etc# postconf -n | grep restrictions
dms_smtpd_sender_restrictions = permit_sasl_authenticated, permit_mynetworks, reject_unknown_sender_domain
mua_sender_restrictions = $dms_smtpd_sender_restrictions
smtpd_client_restrictions = permit_mynetworks, permit_sasl_authenticated, reject_unauth_destination
smtpd_data_restrictions = permit_mynetworks, permit_sasl_authenticated, reject_unauth_pipelining
smtpd_helo_restrictions = permit_mynetworks, reject_invalid_helo_hostname, permit
smtpd_recipient_restrictions = permit_sasl_authenticated, permit_mynetworks, reject_unauth_destination, reject_unauth_pipelining, reject_invalid_helo_hostname, reject_non_fqdn_helo_hostname, reject_unknown_recipient_domain, reject_rbl_client zen.spamhaus.org, reject_rbl_client bl.spamcop.net
smtpd_relay_restrictions = permit_mynetworks permit_sasl_authenticated defer_unauth_destination
smtpd_sender_restrictions = $dms_smtpd_sender_restrictions
```

Thanks for mention the new docs for the proxy protocol! i can remember it was a mess to get this working, but fortunately it works since 2+ years🎉

**Reply by polarathene** [reply]
> Is there are possibility to increase the log level for postfix?

https://www.postfix.org/DEBUG_README.html#debug_peer

You could try the `debug_peer_list = connecting_ip_here` advice, or editing `master.cf` submission service to have `smtpd -v` instead of just `smtpd` as the link shows a few examples further down (_you can do this in the container when it's running, use `nano /etc/postfix/master.cf`, then after saving the change use command `postfix reload`, and try connect from local again_).

Alternatively, you might be able to confirm if adding your local DMS IP to the remote trusted networks would work, `postconf 'mynetworks = local_client_ip_here' && postfix reload`. If that works it is because of the `permit_mynetworks` rule, which means `permit_sasl_authenticated` might not be working correctly (_remember that until v14 of DMS, only the `:edge` build with relay fixes is going to work for you like we discussed earlier, if you switched back to a release version instead of edge, this could be why you have a problem_).

**Reply by dfoxg** [reply]
Thanks! I added my ip to `debug_peer_list ` and `mynetworks`  and got following logs (remote server is on v13.3.1):
```
>>> START Client host RESTRICTIONS <<<
generic_checks: name=permit_sasl_authenticated
generic_checks: name=permit_sasl_authenticated status=0
generic_checks: name=reject
NOQUEUE: reject: RCPT from <my private ip>: 554 5.7.1 <<my private ip>>: Client host rejected: Access denied; from=<d@<my local domain>> to=<test-mail> proto=ESMTP helo=<mail.<my local domain>>
generic_checks: name=reject status=2
>>> END Client host RESTRICTIONS <<<
```

Then i updated the remote mailserver (this instance was on 13.3.1 the whole time, i only updated my local mailserver to edge before) to the latest edge-image, but still got the exact same logs:
```
>>> START Client host RESTRICTIONS <<<
generic_checks: name=permit_sasl_authenticated
generic_checks: name=permit_sasl_authenticated status=0
generic_checks: name=reject
NOQUEUE: reject: RCPT from <my private ip>: 554 5.7.1 <<my private ip>>: Client host rejected: Access denied; from=<d@<my local domain>> to=<test-mail> proto=ESMTP helo=<mail.<my local domain>>
generic_checks: name=reject status=2
>>> END Client host RESTRICTIONS <<<
```

**Reply by polarathene** [reply]
Can you have your mail client connect to port 587 directly on the remote DMS instead of the local DMS? Is that successful to send mail from or also failing?

**Reply by dfoxg** [reply]
Just tested it, works like charm.

**Reply by polarathene** [reply]
Does it still work like a charm if you run that from within the local DMS container?

If so then perhaps review your relay config? Two separate DMS instances were configured to [work correctly with relay host config](https://github.com/docker-mailserver/docker-mailserver/issues/3860) (_really lengthy discussion, but some config snippets can be useful_).

These settings should be set:

`/etc/postfix/sasl_passwd` should contain your credentials to login to remote DMS for relaying.

https://github.com/docker-mailserver/docker-mailserver/blob/a815bf5ab44388baecb597a0b83f71872a2d34bc/target/scripts/helpers/relay.sh#L94-L101

`/etc/postfix/relayhost_map` should contain your remote DMS and port 587 to login:

https://github.com/docker-mailserver/docker-mailserver/blob/a815bf5ab44388baecb597a0b83f71872a2d34bc/target/scripts/helpers/relay.sh#L133

Or since you're using `DEFAULT_RELAY_HOST` ENV, you should be relying on this `main.cf` setting being configured correctly with your remote DMS + port configured:

https://github.com/docker-mailserver/docker-mailserver/blob/a815bf5ab44388baecb597a0b83f71872a2d34bc/target/scripts/helpers/relay.sh#L180-L183

---

The v14 / edge docs for relayhost feature summarizes the settings managed by the feature here:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/697f1086-5b42-45ed-b3bf-3292df42d971)

If you disable any relay host ENV configuration, you can add that via our `postfix-main.cf` override support, this should be all you need:

```
# Send all outbound mail through this relay service:
relayhost = [remote-dms.example.com]:587

# Remote DMS account credentials (static table type): `<username> : <password>`
smtp_sasl_password_maps = static:john.doe@relay-service.com:secret

smtp_sasl_auth_enable = yes
# These two require all outbound mail to be via the relay (which is what the `relayhost` setting above does):
smtp_sasl_security_options = noanonymous
smtp_tls_security_level = encrypt
```

Remember to use `docker compose up --force-recreate` when "restarting" to properly apply config changes.

**Reply by dfoxg** [reply]
I just found my problem.. `/etc/postfix/sasl_passwd` contained a invalid entry, probability from my testing which I didnt cleaned up.. so there was:
```
@<my local domain> <invalid-user>:<invalid-password>
mail.<my remote domain>:587    <valid email>:<valid password>
```

So docker mailserver was using the wrong credentials the whole time :( Sorry for wasting your time with that :/

But just an idea, so that another one isnt running in the same problem: Currently the `setup relay` command only allows adding some entrys, but not viewing/removing entrys. Probably it would be great to have a `setup relay view`-command of something like that?

PS: I just sponsored the project, thanks for your incredible help!

**Reply by polarathene** [reply]
> Sorry for wasting your time with that :/

No worries, got there in the end! Glad it wasn't a new bug :)

> Currently the `setup relay` command only allows adding some entries, but not viewing/removing entries. Probably it would be great to have a `setup relay view`-command of something like that?

There's a related issue for that already I think. The feature needs to be refactored first, something like that could follow afterwards 👍 

---

> PS: I just sponsored the project, thanks for your incredible help!

Awesome, thanks! @georglauterbach will appreciate the support to keep the project going 😁 (_I don't take sponsorship as I plan to step down from my maintainer role in future_)

**Reply by georglauterbach** [reply]
Very much appreciated, thanks!
